tick/optim/model/tests/linreg_with_intercepts_test.py

```diff
@@ -13,27 +13,82 @@
 
 class Test(TestGLM):
     def test_ModelLinRegWithIntercepts(self):
-        """...Numerical consistency check of loss and gradient for Linear
-        Regression
+        """...Numerical consistency check of loss and gradient for linear
+        regression with sample intercepts
         """
-
+        # *** Tests without global intercept ***
         np.random.seed(12)
         n_samples, n_features = 200, 5
         w0 = np.random.randn(n_features)
         intercept0 = 50 * weights_sparse_gauss(n_weights=n_samples, nnz=30)
-        X, y = SimuLinReg(w0, None, n_samples=n_samples,
-                          verbose=False, seed=2038).simulate()
+        c0 = None
+        X, y = SimuLinReg(w0, c0, n_samples=n_samples, verbose=False,
+                          seed=2038).simulate()
         # Add gross outliers to the labels
         y += intercept0
         X_spars = csr_matrix(X)
-        model = ModelLinRegWithIntercepts().fit(X, y)
-        model_spars = ModelLinRegWithIntercepts().fit(X_spars, y)
+        model = ModelLinRegWithIntercepts(fit_intercept=False).fit(X, y)
+        model_spars = ModelLinRegWithIntercepts(fit_intercept=False) \
+            .fit(X_spars, y)
         self.run_test_for_glm(model, model_spars, 1e-4, 1e-4)
 
         self.assertAlmostEqual(model.get_lip_mean(), 6.324960325598532)
         self.assertAlmostEqual(model.get_lip_max(), 30.277118951892113)
         self.assertAlmostEqual(model.get_lip_mean(), model_spars.get_lip_mean())
         self.assertAlmostEqual(model.get_lip_max(), model_spars.get_lip_max())
+        self.assertAlmostEqual(model.get_lip_best(), 2.7217793249045439)
+
+        # *** Tests with global intercept ***
```

Comment:
Again, instead of commenting, it would be better to split this test in two tests